### PR TITLE
Linux memory increased

### PR DESCRIPTION
**Pull request by lvale**
Linux boards don't have the memory constraints that PX family has, so the limits for storage have been increased. It's advisable to use this incremented storage carefully so that GCS can cope with the increased values.
This change will require a fresh configuration of the flight controller if Linux based.

On Linux based boards this is, for now, 200 rally points, 1000 fence points, 7000 waypoints.

Also parameter memory has been increased to 0xF00 on Copter, Plane, Rover

### CODE REVIEW

**Comment by tridge** [comment]
this looks good, but we will need some work here:
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_HAL_Linux/Storage.cpp#L74
to avoid users losing their parameters when they upgrade

**Comment by lvale** [comment]
@lucasdemarchi um pouco de samba sff (a little samba please) :)

**Comment by magicrub** [comment]
Probably need to do something about the merge too.

**Comment by kwikius** [comment]
Is it not possible on PX4 to use the SD-CARD to store this ( and indeed  all param) data so providing "unlimited" numbers independent of memory?  On a new board I was tentatively thinking to use SD_Card  rather than FRAM/EEPROM for storage. Checking that the SD-CARD params and aircraft  id's match by some device ( small eeprom maybe) to avoid flying the wrong sd-card.

I am wondering if that is a bad idea ?

**Comment by lucasdemarchi** [comment]
@kwikius remember that whatever was stored there you will load at runtime. It would be possible to load piece by piece of a mission for example, but it is not as simples as just increasing the space as in Linux.

**Comment by lvale** [comment]
@kwikius That method to read and write to the SD while on a mission was one of the topics we've discussed on a recent dev meeting. Doable, but timing issues could come up. Would be great even to have multiple missions on the card and switch between missions :). Even better would be to have a scripting language for missions :).

This expansion makes sense for Linux boards that were constrained to keep STM compatibility,

Also, Ground Stations must be adapted to cope with these larger numbers.

**Comment by lucasdemarchi** [comment]
LGTM.  Is this tested?

**Comment by lucasdemarchi** [comment]
It needs a rebase and commits to be squashed. I can do that when applying or when someone else applies.

File: libraries/AP_HAL_Linux/Storage.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ void Storage::_storage_open(void)
       without forcing users to reset all parameters
      */
     ssize_t ret = read(fd, _buffer, sizeof(_buffer));
-    if (ret == 4096 && ret != sizeof(_buffer)) {
+    if ((ret == 4096 || ret == 16384) && ret != sizeof(_buffer)) {
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/Storage.cpp]: comment above also needs to be updated.

**Comment by lucasdemarchi** [comment]
new version: #4172

**Pull request closed without merging**